specs/WebResourceRequestedWorkers.md

```diff
@@ -0,0 +1,210 @@
+                {
+                    COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_KINDS requestSourceKinds =
+                        COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_KINDS_ALL;
+                    webResourceRequestArgs->get_RequestSourceKinds(&requestSourceKinds);
+                    COREWEBVIEW2_WEB_RESOURCE_CONTEXT context;
+                    args->get_ResourceContext(&context);
+                    if (requestSourceKinds ==
+                            COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_KINDS_SERVICE_WORKER &&
+                        context == COREWEBVIEW2_WEB_RESOURCE_CONTEXT_IMAGE)
+                    {
+                        Microsoft::WRL::ComPtr<IStream> response_stream;
+                        Microsoft::WRL::ComPtr<ICoreWebView2_2> webview2;
+                        Microsoft::WRL::ComPtr<ICoreWebView2Environment> environment;
+                        CHECK_FAILURE(m_webviewEventSource->QueryInterface(IID_PPV_ARGS(&webview2)));
+                        CHECK_FAILURE(webview2->get_Environment(&environment));
+                        CHECK_FAILURE(SHCreateStreamOnFileEx(
+                            file_path, STGM_READ, FILE_ATTRIBUTE_NORMAL,
+                            FALSE, nullptr, &response_stream));
+                        Microsoft::WRL::ComPtr<ICoreWebView2WebResourceResponse> response;
+                        CHECK_FAILURE(environment->CreateWebResourceResponse(
+                            response_stream.Get(), 200, L"OK", L"Content-Type: image/jpeg",
+                            &response));
+                        args->put_Response(response.Get());
+                    }
+                }
+                return S_OK;
+            })
+            .Get(),
+        &m_webResourceRequestedToken);
+```
+
+# API Details
+## IDL
+```c#
+/// Specifies the source of `WebResourceRequested` event.
+[v1_enum]
+typedef enum COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_KINDS {
+  /// Indicates that web resource is requested from main page including dedicated workers and iframes.
+  COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_KINDS_DOCUMENT = 1,
+
+  /// Indicates that web resource is requested from shared worker.
+  COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_KINDS_SHARED_WORKER = 2,
+
+  /// Indicates that web resource is requested from service worker.
+  COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_KINDS_SERVICE_WORKER = 4,
+
+  /// Indicates that web resource is requested from any supported source.
+  COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_KINDS_ALL = 0XFFFFFFFF
+} COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_KINDS;
+cpp_quote("DEFINE_ENUM_FLAG_OPERATORS(COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_KINDS);")
+
+[uuid(1cc6a402-3724-4e4a-9099-8cf60f2f93a1), object, pointer_default(unique)]
+interface ICoreWebView2_16: ICoreWebView2_15 {
+  /// A web resource request with a resource context that matches this
+  /// filter's resource context, an URI that matches this filter's URI
```

Comment:
"an URI", typo of "and a URI" ?   I think you're mirroring the opening remark from https://docs.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2.addwebresourcerequestedfilter?view=webview2-dotnet-1.0.1264.42

Reply:
Looks like a typo in the existing docs:

|URI Filter String | Request URI | Match | Notes|
|-|-|-|-|
| *example | https://example | No | The URI is normalized before filter matching so the actual URI used for comparison is https://example.com/
|*example/ | https://example | Yes | Just like above, but this time the filter ends with a / just like the normalized URI

I think this should be

|URI Filter String | Request URI | Match | Notes|
|-|-|-|-|
| *example | https://example | No | The URI is normalized before filter matching so the actual URI used for comparison is https://example/
|*example/ | https://example | Yes | Just like above, but this time the filter ends with a / just like the normalized URI